internal/engine/session.go

```diff
@@ -119,25 +142,42 @@ func NewSession(config SessionConfig) (*Session, error) {
 		byteCounter:             bytecounter.New(),
 		kvStore:                 config.KVStore,
 		logger:                  config.Logger,
-		proxyURL:                config.ProxyURL,
 		queryProbeServicesCount: atomicx.NewInt64(),
 		softwareName:            config.SoftwareName,
 		softwareVersion:         config.SoftwareVersion,
 		tempDir:                 tempDir,
 		torArgs:                 config.TorArgs,
 		torBinary:               config.TorBinary,
 	}
+	proxyURL := config.ProxyURL
+	if proxyURL != nil {
+		switch proxyURL.Scheme {
+		case "psiphon", "tor":
+			tunnel, err := tunnel.Start(ctx, tunnel.Config{
+				Name:    proxyURL.Scheme,
+				Session: sess,
```

Comment:
🚨 This code is, sadly, wrong: `tunnel.Start` will ask the session to give it an HTTP transport, which we do not have yet and so we need to start the real tunnel in a deferred way. Or, we need to refactor the tunnel to be lazy. This is a bit of a blocker for moving forward. I need to think about another way or organizing the code.

Reply:
I have fixed this circular dependency now.